packages/design-system/src/powerhouse/components/icon/icon.tsx

```diff
@@ -1,20 +1,61 @@
-import { iconComponents } from "@/assets";
-import { Color, getDimensions, IconName, Size } from "@/powerhouse";
-import { ComponentPropsWithoutRef } from "react";
+import { type IconName, Props } from "@/assets/icon-components/types";
+import { Color, getDimensions, Size } from "@/powerhouse";
+import {
+  ComponentPropsWithoutRef,
+  ComponentType,
+  ElementType,
+  lazy,
+  Suspense,
+  useMemo,
+} from "react";
+
+export { iconNames } from "@/assets/icon-components/types";
+export type { IconName } from "@/assets/icon-components/types";
 
 export type IconProps = ComponentPropsWithoutRef<"svg"> & {
   readonly name: IconName;
   readonly size?: Size;
   readonly color?: Color;
 };
 
+function IconErrorFallback(props: Props) {
+  return <div style={{ width: props.width, height: props.height }} />;
+}
+
+function loadIcon(name: IconName): ElementType {
+  try {
+    return lazy<ComponentType<Props>>(
+      () => import(`@/assets/icon-components/${name}.tsx`),
+    );
+  } catch (e) {
+    console.error(e);
+    return IconErrorFallback;
+  }
+}
+
+export function preloadIcon(name: IconName) {
+  return loadIcon(name);
+}
+
 export function Icon({ name, size = 24, color, style, ...props }: IconProps) {
   const dimensions = getDimensions(size);
   const _style = {
     color,
     ...dimensions,
     style,
   };
-  const IconComponent = iconComponents[name];
-  return <IconComponent {...props} style={_style} />;
+
+  const IconComponent = useMemo(() => loadIcon(name), [name]);
+
+  return (
+    // displays div with the same size while icon
+    // loads to avoid UI displacement
+    <Suspense
+      fallback={<div data-testid="icon-fallback" style={dimensions} />}
```

Comment:
While the icon is loading, a div with the same size is displayed so it takes up the same space in the UI.